### ISSUE TITLE
[BUG] The `AddBuiltInNoProxy` function does not handle whitespace

### ISSUE THREAD

**Issue by votdev** [harvester/harvester#5900]
**Is your enhancement related to a problem? Please describe.**
When a user enters a `noProxy` setting like ` 10.0.0.0/8  ,  127.0.0.1 `, the [built-in noProxy configuration](https://github.com/harvester/harvester/blob/5ee0329f1a6acef48e52758d3caf1d19e3bac1aa/pkg/util/proxy.go#L9) will add duplicates, in this case `127.0.0.1`.

**Describe the solution you'd like**
When splitting the noProxy string for processing, whitespaces need to be removed beforehand. 

**Describe alternatives you've considered**
n/a

**Comment by TachunLin** [comment]
@votdev, 

I tried first on `v1.3.0` Harvester to reproduce the original issue behavior. 
But after I set the noProxy setting in the Harvester proxy settings page  `10.0.0.0/8  ,  127.0.0.1`. 

There we did not find the setting have duplicate `127.0.0.1`.

  ![image](https://github.com/user-attachments/assets/abd37754-02bf-42e8-9d42-8d5d075913e5)
  
  ![image](https://github.com/user-attachments/assets/9012f525-8ed4-4ef7-b566-0a7c4712cda6)


May I know the correct place and steps to reproduce this issue. thank you.

**Comment by votdev** [comment]
> @votdev,
> 
> I tried first on `v1.3.0` Harvester to reproduce the original issue behavior. But after I set the noProxy setting in the Harvester proxy settings page `10.0.0.0/8 , 127.0.0.1`.
> 
> There we did not find the setting have duplicate `127.0.0.1`.
> 
> ![image](https://private-user-images.githubusercontent.com/29251855/353411105-abd37754-02bf-42e8-9d42-8d5d075913e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNDg1OTMsIm5iZiI6MTcyNDA0ODI5MywicGF0aCI6Ii8yOTI1MTg1NS8zNTM0MTExMDUtYWJkMzc3NTQtMDJiZi00MmU4LTlkNDItOGQ1ZDA3NTkxM2U1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDA2MTgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzNDlkZDI4OTM4MzI5YjY0MjVmNzIyMzNhMDU1NTEzZTk0Y2Q0MWYzOGM0NGUwYWQyMmFhY2MwOTRjNmYwODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QvHWkLyTiQoSZAgIsGByW19IScoHLHYi3S-cOS2EBKE)
> 
> ![image](https://private-user-images.githubusercontent.com/29251855/353411185-9012f525-8ed4-4ef7-b566-0a7c4712cda6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNDg1OTMsIm5iZiI6MTcyNDA0ODI5MywicGF0aCI6Ii8yOTI1MTg1NS8zNTM0MTExODUtOTAxMmY1MjUtOGVkNC00ZWY3LWI1NjYtMGE3YzQ3MTJjZGE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE5VDA2MTgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4MTRmMGMzMmEwNDYzOTI4MDE2MjBkYzZlMjk1MGQ5YWZiNDM2ZDE0Y2Q3OGZmNzRjODIwZjgzMmVjNGM5MzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QeCBJi1mvkuTW4DnmMhf_wjXVlPubzawoWBhXbHd5IE)
> 
> May I know the correct place and steps to reproduce this issue. thank you.

The duplication is not visible in the proxy setting itself. The configured `noProxy` setting is used internally in many places (see the [references](https://github.com/harvester/harvester/blob/5ee0329f1a6acef48e52758d3caf1d19e3bac1aa/pkg/util/proxy.go#L28) of the function we are talking about), e.g. you can see the settings synced to `Containers.rancher.Environment.NO_PROXY` of the pod `cattle-system/rancher-xxxx`:
```
Environment:
    CATTLE_NAMESPACE:           cattle-system
    CATTLE_PEER_SERVICE:        rancher
    CATTLE_FEATURES:            multi-cluster-management=false,multi-cluster-management-agent=false
    CATTLE_SYSTEM_CATALOG:      bundled
    CATTLE_BOOTSTRAP_PASSWORD:  <set to the key 'bootstrapPassword' in secret 'bootstrap-secret'>  Optional: false
    HTTP_PROXY:                 
    HTTPS_PROXY:                
    NO_PROXY:                   192.168.0.0/24,127.0.0.1,localhost,0.0.0.0,10.0.0.0/8,longhorn-system,cattle-system,cattle-system.svc,harvester-system,.svc,.cluster.local
```

**Comment by brandboat** [comment]
Test OK in v1.4.0-rc3, will close the issue.

- in v1.3.0, filled `10.0.0.0/8  ,  127.0.0.1` to noProxy setting, duplicate value show up.
<img width="1677" alt="image" src="https://github.com/user-attachments/assets/a4fdadb6-5092-455b-8d98-b52a662f0b80">

- confirmed this was fixed in v1.4.0-rc3
<img width="1669" alt="image" src="https://github.com/user-attachments/assets/7369293f-f4db-4954-9336-d75a8682c8e0">
